pkg/chains/chain.go

```diff
@@ -101,8 +101,8 @@ func IsZetaChain(chainID int64) bool {
 	return ChainIDInChainList(chainID, ChainListByNetwork(Network_zeta))
 }
 
-// IsHeaderSupportedEvmChain returns true if the chain is an EVM chain supporting block header-based verification
-func IsHeaderSupportedEvmChain(chainID int64) bool {
+// IsHeaderSupportedChain returns true if the chain is supports block header-based verification
```

Comment:
```suggestion
// IsHeaderSupportedChain returns true if the chain's consensus supports block header-based verification
```
Deciding if a chain support header is part of the verification flags. The static part here would be more about if the underlying consensus support headers